internal/commands/logs/list.go

```diff
@@ -38,25 +44,79 @@ func (cmd *CommandList) Inputs() cli.InputResolver {
 
 // Handler is the command handler
 func (cmd *CommandList) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Clients) error {
+	cmdStart := time.Now() // for use with tail later
+
 	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
 	if err != nil {
 		return err
 	}
 
-	logs, err := clients.Realm.Logs(app.GroupID, app.ID, realm.LogsOptions{
+	opts := realm.LogsOptions{
 		Types:      cmd.inputs.Types,
 		ErrorsOnly: cmd.inputs.Errors,
-		Start:      cmd.inputs.Start.Time,
-		End:        cmd.inputs.End.Time,
-	})
+	}
+	if !cmd.inputs.Tail {
+		opts.Start = cmd.inputs.Start.Time
+		opts.End = cmd.inputs.End.Time
+	}
+
+	logs, err := clients.Realm.Logs(app.GroupID, app.ID, opts)
 	if err != nil {
 		return err
 	}
 
-	sort.SliceStable(logs, func(i, j int) bool {
-		return logs[i].Started.Before(logs[j].Started)
-	})
+	if cmd.inputs.Tail && len(logs) > tailLookBehind {
+		logs = logs[0:tailLookBehind]
+	}
+
+	printLogs(ui, logs)
+	if !cmd.inputs.Tail {
+		return nil // if not tailing, command stops here
+	}
+
+	logsCh, errCh, closeCh := make(chan realm.Logs), make(chan error, 1), make(chan struct{}, 1)
+
+	opts.Start = cmdStart
+	go pollForLogs(clients.Realm, app.GroupID, app.ID, opts, logsCh, errCh, closeCh)
+
+	for {
+		select {
+		case logs := <-logsCh:
+			printLogs(ui, logs)
+		case err := <-errCh:
```

Comment:
do we need to `close(closeCh)` here too?

Reply:
yeah, probably.  i'll add it as a defer above all of this